From 5a807d30e2ba27f901d845f6e1d622390ce6d63e Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Wed, 8 Oct 2014 15:29:12 -0700
Subject: [PATCH 176/290] drm: xilinx: drv: Add xilinx_drm_get_format()
 function

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

The function retreives the current DRM format of the device.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 2f7ab1b3b648ad3a9b296e31f49c5cfa98571c70)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_drv.c | 15 +++++++++++++++
 drivers/gpu/drm/xilinx/xilinx_drm_drv.h |  1 +
 2 files changed, 16 insertions(+)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_drv.c b/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
index 67ce9e4..901fe32 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
@@ -94,6 +94,21 @@ bool xilinx_drm_check_format(struct drm_device *drm, uint32_t fourcc)
 	return xilinx_drm_crtc_check_format(private->crtc, fourcc);
 }
 
+/**
+ * xilinx_drm_get_format - Get the current device format
+ * @drm: DRM device
+ *
+ * Get the current format of pipeline
+ *
+ * Return: the corresponding DRM_FORMAT_XXX
+ */
+uint32_t xilinx_drm_get_format(struct drm_device *drm)
+{
+	struct xilinx_drm_private *private = drm->dev_private;
+
+	return xilinx_drm_crtc_get_format(private->crtc);
+}
+
 /* create a fb */
 static struct drm_framebuffer *
 xilinx_drm_fb_create(struct drm_device *drm, struct drm_file *file_priv,
diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_drv.h b/drivers/gpu/drm/xilinx/xilinx_drm_drv.h
index 5e95b5a..fbb62fd 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_drv.h
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_drv.h
@@ -45,5 +45,6 @@ static inline u32 xilinx_drm_readl(void __iomem *base, int offset)
 struct drm_device;
 
 bool xilinx_drm_check_format(struct drm_device *drm, uint32_t fourcc);
+uint32_t xilinx_drm_get_format(struct drm_device *drm);
 
 #endif /* _XILINX_DRM_H_ */
-- 
2.0.2

